lib/logstash/outputs/mongodb.rb

```diff
@@ -46,19 +46,22 @@ class LogStash::Outputs::Mongodb < LogStash::Outputs::Base
   # Mutex used to synchronize access to 'documents'
   @@mutex = Mutex.new
 
-  public
   def register
+    if @bulk_size > 1000
+      raise LogStash::ConfigurationError, "Bulk size must be lower than '1000', currently '#{@bulk_size}'"
+    end
+
     Mongo::Logger.logger = @logger
     conn = Mongo::Client.new(@uri)
     @db = conn.use(@database)
 
-    if @bulk_size > 1000
-      raise LogStash::ConfigurationError, "Bulk size must be lower than '1000', currently '#{@bulk_size}'"
-    end
+    @closed = Concurrent::AtomicBoolean.new(false)
     @documents = {}
-    Thread.new do
+
+    @bulk_thread = Thread.new(@bulk_interval) do |bulk_interval|
```

Comment:
if `@bulk` is false, we should avoid creating the bulk thread.

Reply:
agree, but I wanted to minimize functional changes and figured this will anyway be revisited when refactoring the concurrency control code (and that thread has very minimal impact)